director_test.go

```diff
@@ -0,0 +1,192 @@
+		ReadHTTPS: true,
+		UseTokenOnRead: true,
+	}
+
+	// Call the function in question
+	var ns Namespace
+	err := CreateNsFromDirectorResp(directorResponse, &ns)
+
+	// Test for expected outputs
+	assert.NoError(t, err, "Error creating Namespace from Director response")
+
+	assert.Equal(t, constructedNamespace.SortedDirectorCaches, ns.SortedDirectorCaches)
+	assert.Equal(t, constructedNamespace.Path, ns.Path)
+	assert.Equal(t, constructedNamespace.Issuer, ns.Issuer)
+	assert.Equal(t, constructedNamespace.ReadHTTPS, ns.ReadHTTPS)
+	assert.Equal(t, constructedNamespace.UseTokenOnRead, ns.UseTokenOnRead)
+}
+
+func TestNewTransferDetailsUsingDirector(t *testing.T) {
+	os.Setenv("http_proxy", "http://proxy.edu:3128")
+
+	// Construct the input caches
+	// Cache with http
+	nonAuthCache := DirectorCache{
+		ResourceName: "mycache",
+		EndpointUrl: "my-cache-url:8000",
+		Priority: 99,
+		AuthedReq: false,
+	}
+
+	// Cache with https
+	authCache := DirectorCache{
+		ResourceName: "mycache",
+		EndpointUrl: "my-cache-url:8443",
+		Priority: 99,
+		AuthedReq: true,
+	}
+
+	// Case 1: cache with http
+	transfers := NewTransferDetailsUsingDirector(nonAuthCache, nonAuthCache.AuthedReq)
+	assert.Equal(t, 2, len(transfers))
+	assert.Equal(t, "my-cache-url:8000", transfers[0].Url.Host)
+	assert.Equal(t, "http", transfers[0].Url.Scheme)
+	assert.Equal(t, true, transfers[0].Proxy)
+	assert.Equal(t, 2, len(transfers))
+	assert.Equal(t, "my-cache-url:8000", transfers[1].Url.Host)
+	assert.Equal(t, "http", transfers[1].Url.Scheme)
+	assert.Equal(t, false, transfers[1].Proxy)
+
+	// Case 2: cache with https
+	transfers = NewTransferDetailsUsingDirector(authCache, authCache.AuthedReq)
+	assert.Equal(t, 1, len(transfers))
+	assert.Equal(t, "my-cache-url:8443", transfers[0].Url.Host)
+	assert.Equal(t, "https", transfers[0].Url.Scheme)
+	assert.Equal(t, false, transfers[0].Proxy)
+
+	// Case 3: cache without port with http
+	nonAuthCache.EndpointUrl = "my-cache-url"
+	transfers = NewTransferDetailsUsingDirector(nonAuthCache, nonAuthCache.AuthedReq)
+	assert.Equal(t, 2, len(transfers))
+	assert.Equal(t, "my-cache-url:8000", transfers[0].Url.Host)
+	assert.Equal(t, "http", transfers[0].Url.Scheme)
+	assert.Equal(t, true, transfers[0].Proxy)
+	assert.Equal(t, "my-cache-url:8000", transfers[1].Url.Host)
+	assert.Equal(t, "http", transfers[1].Url.Scheme)
+	assert.Equal(t, false, transfers[1].Proxy)
+
+	// Case 4. cache without port with https
+	authCache.EndpointUrl = "my-cache-url"
+	transfers = NewTransferDetailsUsingDirector(authCache, authCache.AuthedReq)
+	assert.Equal(t, 2, len(transfers))
+	assert.Equal(t, "my-cache-url:8444", transfers[0].Url.Host)
+	assert.Equal(t, "https", transfers[0].Url.Scheme)
+	assert.Equal(t, false, transfers[0].Proxy)
+	assert.Equal(t, "my-cache-url:8443", transfers[1].Url.Host)
+	assert.Equal(t, "https", transfers[1].Url.Scheme)
+	assert.Equal(t, false, transfers[1].Proxy)
+}
+
+func TestQueryDirector(t *testing.T) {
```

Comment:
This function is using 4 spaces rather than a tab.  Fix the indentation, and I think we're good.